lib/pdf/core/text.rb

```diff
@@ -306,6 +285,76 @@ def add_text_content(text, x, y, options)
 
         add_content "ET\n"
       end
+
+      private
+
+      def wrap_and_restore_text_rendering_mode(block_value)
+        original_value = text_rendering_mode
+        @text_rendering_mode = block_value
+        update_text_rendering_mode_state
+        begin
+          yield
+        ensure
+          @text_rendering_mode = original_value
+          update_text_rendering_mode_state
+        end
+      end
+
+      def update_text_rendering_mode_state
+        add_content "\n#{MODES[text_rendering_mode]} Tr"
+      end
```

Comment:
This feels like DRY overdone. Inlining it is fine since it's used from within only one method.